File: server-go-dgraph/dql/notification/query.go
package status

// 查询回复通知列表
var Notification_Replies = `query all(
  $loggedUserid: string,
  $first: string, 
  $after: string
) {
  notifications(
		func: type(Notification),
		orderdesc: created_at,
		first: $first,
		after:$after
	) @filter(uid_in(recipient,$loggedUserid) AND eq(action,0)) {
		id:uid        
		sender{
			id:uid        
			name
			username
			verify_name
			avatar_url
			verified
		}
		created_at
		target_type
		target{
			id:uid        
			text
			media_type
			user{
				name
				username
				verify_name
				avatar_url
				verified
			}
			to_user{
				name
				username
				verify_name
			}
			urls
			images{
				url
				source
				platform
			}
			hashtags{
				id:uid
				name
				description
			}
			video{
				url
				source
				platform
			}
			reply_count
			quote_count
			restatus_count
			favorite_count
			platform

			favorite_cnt as fcnt:count(~favorites @filter(uid($loggedUserid)))
			favorited: math(favorite_cnt == 1)

			restatus_cnt as rcnt:count(~restatuses @filter(uid($loggedUserid)))
			restatused: math(restatus_cnt == 1)

			created_at
			to_reply{
				id:uid        
				text
				user{
					name
					verify_name
				}
			}
		}
  }
}`

// 查询点赞通知列表
var Notification_Favorites = `query all(
  $loggedUserid: string,
  $first: string, 
  $after: string
) {
  notifications(
		func: type(Notification),
		orderdesc: created_at, 
		first: $first, 
    after:$after
	) @filter(uid_in(recipient,$loggedUserid) AND eq(action,1)) {
		id:uid        
		sender{
			id:uid        
			name
			username
			verify_name
			avatar_url
			verified
		}
		created_at
		target_type
		target{
			id:uid        
			text
			to_user{
				name
				verify_name
			}
			hashtags{
				id:uid
				name
				description
			}
		}
  }
}`

// 查询转帖通知列表
var Notification_Restatus = `query all(
  $loggedUserid: string,
  $first: string, 
  $after: string
) {
  notifications(
		func: type(Notification),
		orderdesc: created_at, 
		first: $first, 
    after:$after
	) @filter(uid_in(recipient,$loggedUserid) AND eq(action,2)) {
		id:uid        
		sender{
			id:uid        
			name
			username
			verify_name
			avatar_url
			verified
		}
		created_at
		target_type
		target{
			id:uid        
			text
			to_user{
				name
				verify_name
			}
			hashtags{
				id:uid
				name
				description
			}
		}
  }
}`

// 查询引用通知列表
var Notification_Quotes = `query all(
  $loggedUserid: string,
  $first: string, 
  $after: string
) {
  notifications(
		func: type(Notification),
		orderdesc: created_at, 
		first: $first, 
    after:$after
	) @filter(uid_in(recipient,$loggedUserid) AND eq(action,3)) {
		id:uid        
		created_at
		target_type
		target{
			id:uid        
			text
			media_type
			user{
				name
				username
				verify_name
				avatar_url
				verified
			}
			urls
			images{
				url
				source
				platform
			}
			hashtags{
				id:uid
				name
				description
			}
			video{
				url
				source
				platform
			}
			reply_count
			quote_count
			restatus_count
			favorite_count
			platform

			favorite_cnt as fcnt:count(~favorites @filter(uid($loggedUserid)))
			favorited: math(favorite_cnt == 1)

			restatus_cnt as rcnt:count(~restatuses @filter(uid($loggedUserid)))
			restatused: math(restatus_cnt == 1)

			created_at
			to_quote{
				id:uid        
				text
				user{
					name
					verify_name
				}
			}
		}
  }
}`

// 查询提及通知列表
var Notification_Mentions = `query all(
  $loggedUserid: string,
  $first: string, 
  $after: string
) {
  notifications(
		func: type(Notification),
		orderdesc: created_at, 
		first: $first, 
    after:$after
	) @filter(uid_in(recipient,$loggedUserid) AND eq(action,4)) {
		id:uid        
		created_at
		target_type
		target{
			id:uid        
			text
			media_type
			user{
				name
				username
				verify_name
				avatar_url
				verified
			}
			urls
			images{
				url
				source
				platform
			}
			hashtags{
				id:uid
				name
				description
			}
			video{
				url
				source
				platform
			}
			reply_count
			quote_count
			restatus_count
			favorite_count
			platform

			favorite_cnt as fcnt:count(~favorites @filter(uid($loggedUserid)))
			favorited: math(favorite_cnt == 1)

			restatus_cnt as rcnt:count(~restatuses @filter(uid($loggedUserid)))
			restatused: math(restatus_cnt == 1)

			created_at
		}
  }
}`

// 查询用户通知列表
var Notification_Users = `query all(
  $loggedUserid: string,
  $first: string, 
  $after: string
) {
  notifications(
		func: type(Notification),
		orderdesc: created_at, 
		first: $first, 
    after:$after
	) @filter(uid_in(recipient,$loggedUserid) AND eq(action,5)) {
		id:uid        
		action
		created_at
		target_type
		target{
			id:uid        
			name
			username
			verify_name
			avatar_url
			verified
		}
  }
}`

// 查询广播通知列表
var Notification_Broadcast = `query all(
  $loggedUserid: string,
  $first: string, 
  $after: string
) {
  notifications(
		func: type(Notification),
		orderdesc: created_at, 
		first: $first, 
    after:$after
	) @filter(uid_in(recipient,$loggedUserid) AND eq(action,6)) {
		id:uid        
		action
		msg
		created_at
  }
}`

File: server-rust-dgraph/src/api/group/dgraph/query.rs

// 查询用户信息
pub const STATUS_DETAIL: &'static str = r#"query all(
  $uid: string,
  $logged_user_id: string
) {
  data(func: uid($uid)) {
    uid 
    text
    is_forward
    replies_count
    forwards_count
    favorites_count
    created_at
    group{
      uid
      name
      is_verified
    }
    user{
      uid
      name
      username
      avatar_url
      description
      is_verified
      verified{
        uid
        name
        description
      }
      cnt as count(~follows @filter(uid($logged_user_id)))
      following: math(cnt == 1)
    }
    entities:{
      urls{
        url
        url_key
      }
      mentions{
        uid
        name
        username
      },
      hashtags{
        uid
        name
      }
    }
    medias{
      photos{
        url
      }
      video{
        url
      }
      music{
        url
      }
    }
  }
}"#;


/*
  1，功能：小组详情
*/ 
pub const GROUP_DETAIL: &'static str = r#"query all(
  $uid: string,
  $logged_user_id: string
) {
  data(func: uid($uid)) {
    uid
    group_name
    group_description
    announcement
    avatar_url
    cover_url
    default_cover
    access
    visible
    members_count
    statuses_count
    created_at

    cnt as count(members @filter(uid_in(user,$logged_user_id)))
    is_joined: math(cnt == 1)

    is_verified
    verified{
      uid
      name
      description
    }    
    group_entities{
      uid
      urls{
        uid
        url
        url_key
      }
      mentions{
        uid
        name
        username
      },
      hashtags{
        uid
        name
      }
      medias{
        uid
        url
        media_type
      }
    }
  }
}"#;


/*
  1，功能：小组{id}的帖子列表
*/ 
pub const GROUP_STATUS: &'static str = r#"query all(
  $uid: string,
  $logged_user_id: string,
  $first: int,
  $after: string
){
  data(func: uid($uid)) {
    uid
    edges: ~group @filter(type(Status)) (first: $first, after: $after) {
      uid 
      text
      is_forward

      reply_cnt as count(replies @filter(uid_in(replies,$logged_user_id)))
      is_replied: math(reply_cnt == 1)
      replies_count

      forward_cnt as count(forwards @filter(uid($logged_user_id)))
      is_forwarded: math(forward_cnt == 1)
      forwards_count

      favorite_cnt as count(favorites @filter(uid($logged_user_id)))
      is_favorited: math(favorite_cnt == 1)
      favorites_count

      created_at
      group{
        uid
        name
        is_verified
      }
      user{
        uid
        name
        username
        avatar_url
        description
        is_verified
        verified{
          uid
          name
          description
        }
        cnt as count(~follows @filter(uid($logged_user_id)))
        following: math(cnt == 1)
      }
      entities{
        uid
        urls{
          uid
          url
          url_key
        }
        mentions{
          uid
          name
          username
        },
        hashtags{
          uid
          name
        }
        medias{
          uid
          url
          media_type
        }
      }
      forward_to_status{
        uid
        text
        created_at
        user{
          uid
          name
          username
          avatar_url
          description
          is_verified
          verified{
            uid
            name
            description
          }
        }
        entities{
          uid
          urls{
            uid
            url
            url_key
          }
          mentions{
            uid
            name
            username
          },
          hashtags{
            uid
            name
          }
          medias{
            uid
            url
            media_type
          }
        }
      }
    }
  }
}"#;